cmd/nvidia-dra-controller/imex.go

```diff
@@ -249,10 +266,10 @@ func (m *ImexManager) streamImexDomains(ctx context.Context) (<-chan string, <-c
 }
 
 // generateImexChannelPool generates the contents of a ResourceSlice pool for a given IMEX domain.
-func generateImexChannelPool(imexDomain string, numChannels int) resourceslice.Pool {
-	// Generate dchannels from 0 to numChannels
+func generateImexChannelPool(imexDomain string, cliqueid int) resourceslice.Pool {
+	// Generate dchannels from ImexChannelLimit*(cliqueid-1) to ImexChannelLimit*(cliqueid)
 	var devices []resourceapi.Device
-	for i := 0; i < numChannels; i++ {
+	for i := (ImexChannelLimit * (cliqueid - 1)); i < ImexChannelLimit*(cliqueid); i++ {
```

Comment:
As discussed in the call the other day, we can't rely solely on the `clique` id to do this. The one and only clique id could be `4294967295` (and in fact it is on one of my test clusters).